graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -91,6 +99,52 @@ protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest request) th
             return unsupportedMediaType();
         }
 
+        // See https://github.com/jaydenseric/graphql-multipart-request-spec/blob/master/readme.md
+        if (contentType.is(MediaType.MULTIPART_FORM_DATA)) {
+            return HttpResponse.from(FileAggregatedMultipart.aggregateMultipart(ctx, request)
+                                                            .thenApply(multipart -> {
+                try {
+                    final String operationsParam = toStringFromMultipartParam("operations", multipart.params());
+                    final String mapParam = toStringFromMultipartParam("map", multipart.params());
+
+                    final Map<String, Object> operation = parseJsonString(operationsParam, JSON_MAP);
+                    final Map<String, List<String>> map = parseJsonString(mapParam, MAP_PARAM);
+                    final String query = toStringFromJson("query", operation.get("query"));
+                    if (Strings.isNullOrEmpty(query)) {
+                        return HttpResponse.of(HttpStatus.BAD_REQUEST, MediaType.PLAIN_TEXT,
+                                               "Missing query");
+                    }
+                    final String operationName =
+                            toStringFromJson("operationName", operation.get("operationName"));
+                    final Map<String, Object> variables = toMapFromJson(operation.get("variables"));
+                    final Map<String, Object> extensions = toMapFromJson(operation.get("extensions"));
+                    for (Map.Entry<String, List<String>> entry : map.entrySet()) {
+                        final String key = entry.getKey();
+                        final List<String> value = entry.getValue();
+                        final List<Path> paths = multipart.files().get(key);
+                        if (!paths.isEmpty()) {
+                            for (int i = 0; i < value.size(); i++) {
+                                MultipartVariableMapper.mapVariable(value.get(i), variables, paths.get(i));
+                            }
+                        }
+                    }
+
+                    return executeGraphql(ctx, GraphqlRequest.of(query, operationName,
+                                                                 variables, extensions));
+                } catch (JsonProcessingException ex) {
+                    return HttpResponse.of(HttpStatus.BAD_REQUEST, MediaType.PLAIN_TEXT,
+                                           "Failed to parse a JSON document");
+                } catch (IllegalArgumentException ex) {
+                    final HttpResponse response = HttpResponse.of(HttpStatus.BAD_REQUEST,
+                                                                  MediaType.PLAIN_TEXT, ex.getMessage());
```

Comment:
Could you handle the case where `ex.getMessage()` returns `null`?

Reply:
This is the same even if it is not a multipart request, so we need to think about it together. `Unknown`, `Client Error`, `Bad Request`?? 🤔

Reply:
Could you elaborate a little bit more?

Reply:
I mentioned it because I thought the same should be done [here](https://github.com/line/armeria/blob/master/graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java#L122...L126).

And if it's null, we've been thinking about what to put in the message.
- `Unknown`
- `Failed processing GraphQL request`
- `Bad Request`
- ...

Reply:
I see. How about just `HttpStatus.BAD_REQUEST.reasonPhrase()`?